File: frontend/src/types/ssl-check.ts
/** Status of the SSL certificate check. */
export type SSLCheckStatus =
  | "valid"
  | "expired"
  | "not_yet_valid"
  | "invalid"
  | "self_signed"
  | "error";

/** Issuer information from the SSL certificate. */
export interface SSLIssuer {
  organization: string;
  commonName: string;
}

/** Parsed SSL certificate details returned by the backend. */
export interface SSLCertificate {
  commonName: string;
  sans: string[];
  issuer: SSLIssuer;
  validFrom: string;
  validUntil: string;
  daysUntilExpiry: number;
  serialNumber: string;
  signatureAlgorithm: string;
  protocol: string;
}

/** A single certificate in the chain of trust. */
export interface SSLChainEntry {
  subject: string;
  issuer: string;
  validFrom: string;
  validUntil: string;
}

/** Full response from GET /api/ssl-check?domain={domain}. */
export interface SSLCheckResponse {
  domain: string;
  valid: boolean;
  status: SSLCheckStatus;
  certificate?: SSLCertificate;
  chain?: SSLChainEntry[];
  error?: string;
}

File: backend/internal/model/sslcheck.go
package model

import "time"

// SSLCheckResponse is the response envelope for the SSL check endpoint.
type SSLCheckResponse struct {
	Domain      string           `json:"domain"`
	Valid       bool             `json:"valid"`
	Status      string           `json:"status"`
	Certificate *SSLCertificate  `json:"certificate,omitempty"`
	Chain       []SSLChainEntry  `json:"chain,omitempty"`
	Error       string           `json:"error,omitempty"`
}

// SSLCertificate holds the parsed certificate details.
type SSLCertificate struct {
	CommonName         string    `json:"commonName"`
	SANs               []string  `json:"sans"`
	Issuer             SSLIssuer `json:"issuer"`
	ValidFrom          time.Time `json:"validFrom"`
	ValidUntil         time.Time `json:"validUntil"`
	DaysUntilExpiry    int       `json:"daysUntilExpiry"`
	SerialNumber       string    `json:"serialNumber"`
	SignatureAlgorithm string    `json:"signatureAlgorithm"`
	Protocol           string    `json:"protocol"`
}

// SSLIssuer holds issuer information from a certificate.
type SSLIssuer struct {
	Organization string `json:"organization"`
	CommonName   string `json:"commonName"`
}

// SSLChainEntry represents a single certificate in the chain.
type SSLChainEntry struct {
	Subject    string    `json:"subject"`
	Issuer     string    `json:"issuer"`
	ValidFrom  time.Time `json:"validFrom"`
	ValidUntil time.Time `json:"validUntil"`
}
